portfolio/src/main/webapp/script.js

```diff
@@ -155,64 +165,114 @@ function createComment(comment) {
         aria-labelledby="rec-${comment.id}"
       >
         <div class="rec-description">
-          <div class="rec-description-txt">
-            ${comment.description || 'No Description.'}
+          <div class="rec-description-txt">${
+            comment.description || 'No Description.'
+          }</div>
+          <div class="row">
+            <a
+              href=${comment.link}
+              target="_blank"
+              rel="noopener noreferrer"
+              class=${linkClass}
+              >Learn more</a
+            >
+            <i class="fa fa-ban col-2" id="delete-btn-${comment.id}"></i>
           </div>
-          <a
-            href=${comment.link}
-            target="_blank"
-            rel="noopener noreferrer"
-            class=${linkClass}
-            >Learn more</a
-          >
         </div>
       </div>
     </div>
   `);
 }
 
+let comments = [];
+let commentIds = [];
+
 /**
- * fetch text from /data to display
+ * Add event listener to detect when comment is being collapsed/expanded in order to rotate caret accordingly
+ * @param {string} cid comment id used to distinguish each unique comment
  */
-function getFetchRequest() {
-  const comments = [];
-  const commentIds = [];
-  fetch('/data')
+function addRotationEvent(cid) {
+  const $elem = $(`#${cid}`);
+  $elem.on('click', () => {
+    $elem.find('.fa').toggleClass('rotated');
+    const angle = $elem.find('.fa').hasClass('rotated') ? 90 : 0;
+    $elem.find('.fa').animate(
+      { deg: angle },
+      {
+        duration: 200,
+        step: function (now) {
+          $elem.find('.fa').css({
+            transform: 'rotate(' + now + 'deg)',
+          });
+        },
+      }
+    );
+  });
+}
+
+/**
+ * Delete the comment selected from datastore, reload existing comments
+ * @param {string} cid comment id used to distinguish each unique comment
+ */
+function deleteComment(cid) {
+  const id = getCommendId(cid);
+  $.ajax({
+    type: 'POST',
+    url: `/delete-data?id=${id}`,
+    success: () => loadComments('reload'),
+  });
+}
+
+/**
+ * fetch comments from datastore to display
+ * @param {string} type the request parameter (load | reload | append)
+ */
+function loadComments(type = 'load') {
+  const numComments = comments.length - 1;
+  if (type == 'load' || type == 'reload') {
+    comments = [];
+    commentIds = [];
+  }
+  fetch(`/data?type=${type}&numComments=${numComments}`)
```

Comment:
another option is to just not send numComments ever. When deleting a comment, we send the delete request to the server and ignore the response content, but we just remove the clicked comment via jQuery. It's a bit odd to send numComments to the server, and have the server just ignore it unless the type is 'reload'.
But that's a lot of work and I think this is fine as is.

Reply:
I also thought about that, but the tutorial said to use the server as a source of truth to avoid the front end being off sync. But it also turned out to be nice having reload specify the number of comments, since I was able to change my code to reload the same comments after a new entry.